controllers/reconcile.go

```diff
@@ -141,8 +188,8 @@ func (r *MicroK8sControlPlaneReconciler) reconcileMachines(ctx context.Context,
 		logger.Info("scaling up control plane")
 
 		return r.bootControlPlane(ctx, cluster, mcp, controlPlane, false)
-	// We are scaling down
-	case numMachines > desiredReplicas:
+	// We are scaling down or rolling out a new version
+	case numMachines > desiredReplicas || len(oldVersionMachines) > 0:
```

Comment:
Unfortunately this version allows multiple nodes to be provisioned at the same time while others get deleted:
```
NAME                             CLUSTER    NODENAME         PROVIDERID                              PHASE         AGE     VERSION
capi-aws-md-0-7cb6d5645b-4pvjs   capi-aws   ip-10-0-15-212   aws:///us-east-1a/i-0e7cd52289ab3ae79   Running       42m     v1.23.0
capi-aws-md-0-7cb6d5645b-6wr2q   capi-aws   ip-10-0-5-114    aws:///us-east-1a/i-02f129f625277ef64   Running       42m     v1.23.0
capi-aws-control-plane-lj4zr     capi-aws   ip-10-0-6-166    aws:///us-east-1a/i-0bfd63ec00c151a93   Running       19m     v1.24.0
capi-aws-control-plane-85h4v     capi-aws                    aws:///us-east-1c/i-033f4e5772802e322   Provisioned   5m5s    v1.25.0
capi-aws-control-plane-nvrwp     capi-aws                    aws:///us-east-1b/i-04f12ec0a75f3cf2f   Provisioned   3m59s   v1.25.0
capi-aws-control-plane-gc52l     capi-aws   ip-10-0-39-108   aws:///us-east-1c/i-0a34c8401143ffeda   Deleting      23m     v1.24.0
```